### PR TITLE
feat: add variant_id to seat info for ExecEd courses

### PR DESCRIPTION
**Pull request by AfaqShuaib09**
## Description

In continue to the PR: https://github.com/openedx/ecommerce/pull/3794
This PR adds the functionality to add `variant_id` to `seat_info` for execEd courses via publication endpoint

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/courses/models.py [hunk 1]
```diff
@@ -155,6 +155,7 @@ def create_or_update_seat(
             remove_stale_modes=True,
             create_enrollment_code=False,
             sku=None,
+            variant_id=None,
```
Comment (DawoudSheraz) [inline ecommerce/courses/models.py:158]: please update docstring to add this argument info.

**Review by DawoudSheraz — APPROVED** [review]
Codewise, it looks alright. I am going to rely on owner and/or Enterprise for better context here.

**Review by Ali-D-Akbar — APPROVED** [review]

**Comment by zawan-ila** [comment]
Have you verified that this works as intended? I believe you need to define the new attribute `variant_id` on the `Seat` product to ensure it gets into the database? As it stands, I do not think the attribute is really being set in the db 🤔

**Comment by zawan-ila** [comment]
If I am correct here^, I expect we'll need something like [3782](https://github.com/openedx/ecommerce/pull/3782/files)

**Comment by AfaqShuaib09** [comment]
> Have you verified that this works as intended? I believe you need to define the new attribute `variant_id` on the `Seat` product to ensure it gets into the database? As it stands, I do not think the attribute is really being set in the db 🤔

Not this time, but during my work on [this pull request](https://github.com/openedx/course-discovery/pull/4186), I had tested these changes with the value of `course_metadata.is_course_run_variant_id_ecommerce_consumable` set to true, and everything was functioning correctly.

image

[thinking] It seems like it is currently adding `variant_id` Product Attribute that is linked with `Course Entitlements` for seats Products as well. Otherwise it'll raise error in unittest too.
I believe it would be better to add it separately for Seat `Product Type`, just as you mentioned, as shown below for other Product Attributes:
image

Nice catch 💯

**Review by iloveagent57 — APPROVED** [review]

**Review by christopappas — APPROVED** [review]

**Pull request merged**